src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/ECDsaOpenSsl.ImportExport.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class ECDsaOpenSsl : ECDsa
+    {
+        /// <summary>
+        ///         ImportParameters will replace the existing key that ECDsaOpenSsl is working with by creating a
+        ///         new key. If the parameters contains only Q, then only a public key will be imported.
+        ///         If the parameters also contains D, then a full key pair will be imported. 
+        ///         The parameters Curve value specifies the type of the curve to import.
+        /// </summary>
+        /// <param name="parameters">The curve parameters.</param>
+        /// <exception cref="CryptographicException">
+        ///     if <paramref name="parameters" /> does not contain valid values.
+        /// </exception>
+        /// <exception cref="NotSupportedException">
+        ///     if <paramref name="parameters" /> references a curve that cannot be imported.
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        ///     if <paramref name="parameters" /> references a curve that is not supported by this platform.
+        /// </exception>
+        public override void ImportParameters(ECParameters parameters)
+        {
+            SafeEcKeyHandle key;
+
+            parameters.Validate();
+
+            if (parameters.Curve.IsPrime)
+            {
+                key = ImportPrimeCurveParameters(parameters);
+            }
+            else if (parameters.Curve.IsCharacteristic2)
+            {
+                key = ImportCharacteristic2CurveParameters(parameters);
+            }
+            else if (parameters.Curve.IsNamed)
+            {
+                key = ImportNamedCurveParameters(parameters);
+            }
+            else
+            {
+                // Implicit case
+                throw new NotSupportedException(string.Format(SR.Cryptography_InvalidCurve, parameters.Curve.CurveType.ToString()));
```

Comment:
Why is this not "PlatformNotSupportedException"?  I had the same thought in CNG, but didn't ask then.  Right now the else is "is implicit", but this would trip for any new curve type.